src/Utility/UtilLib/CMakeLists.txt

```diff
@@ -1,2 +1,4 @@
 ##### Create library and add to the list of compiled libraries
 add_library(utillib schism_geometry.f90 compute_zcor.f90 extract_mod.f90 extract_mod2.f90 stripesearch_unstr.f90 argparse.f90 pt_in_poly_test.f90 interpolate_unstructured_module.f90)
+
+target_link_libraries(utillib ${NetCDFLIBS} ${HDF5_LIBRARIES})
```

Comment:
The explicit dependencies for each library are necessary for dynamic library builds, at least on macOS.